src/lib/resolver.js

```diff
@@ -130,8 +131,9 @@ function resolveSelector(ctx, lang, args, expr, index) {
     return selector;
   }
 
-  const argValue = index[0].args ?
-    resolveIdentifier(ctx, lang, args, index[0].args[0].name)[1] : undefined;
+  const argName = index[0].args ? index[0].args[0].name : index[1];
+  const argValue = argName ?
+    resolveIdentifier(ctx, lang, args, argName)[1] : undefined;
```

Comment:
Let's fork it;  we won't be adding new features to the legacy resolver so it should be easy to maintain it.  OTOH, like you said, supporting both ASTs in one is going to get harder and harder